android/src/main/kotlin/com/linusu/flutter_web_auth/FlutterWebAuthPlugin.kt

```diff
@@ -3,72 +3,61 @@ package com.linusu.flutter_web_auth
 import android.content.Context
 import android.content.Intent
 import android.net.Uri
-
 import androidx.browser.customtabs.CustomTabsIntent
-
 import io.flutter.embedding.engine.plugins.FlutterPlugin
-import io.flutter.plugin.common.BinaryMessenger
 import io.flutter.plugin.common.MethodCall
 import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
-import io.flutter.plugin.common.PluginRegistry.Registrar
 
-class FlutterWebAuthPlugin(private var context: Context? = null, private var channel: MethodChannel? = null): MethodCallHandler, FlutterPlugin {
-  companion object {
-    val callbacks = mutableMapOf<String, Result>()
+class FlutterWebAuthPlugin : MethodCallHandler, FlutterPlugin {
+    private var context: Context? = null
+    private lateinit var channel: MethodChannel
 
-    @JvmStatic
-    fun registerWith(registrar: Registrar) {
-        val plugin = FlutterWebAuthPlugin()
-        plugin.initInstance(registrar.messenger(), registrar.context())
+    companion object {
+        val callbacks = mutableMapOf<String, Result>()
     }
 
-  }
-
-  fun initInstance(messenger: BinaryMessenger, context: Context) {
-      this.context = context
-      channel = MethodChannel(messenger, "flutter_web_auth")
-      channel?.setMethodCallHandler(this)
-  }
-
-  override public fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
-      initInstance(binding.getBinaryMessenger(), binding.getApplicationContext())
-  }
-
-  override public fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
-      context = null
-      channel = null
-  }
-
-  override fun onMethodCall(call: MethodCall, resultCallback: Result) {
-    when (call.method) {
-        "authenticate" -> {
-          val url = Uri.parse(call.argument("url"))
-          val callbackUrlScheme = call.argument<String>("callbackUrlScheme")!!
-          val preferEphemeral = call.argument<Boolean>("preferEphemeral")!!
-
-          callbacks[callbackUrlScheme] = resultCallback
-
-          val intent = CustomTabsIntent.Builder().build()
-          val keepAliveIntent = Intent(context, KeepAliveService::class.java)
+    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+        context = binding.applicationContext
+        channel = MethodChannel(binding.binaryMessenger, "flutter_web_auth")
+        channel.setMethodCallHandler(this)
+    }
 
-          intent.intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
-          if (preferEphemeral) {
-              intent.intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY)
-          }
-          intent.intent.putExtra("android.support.customtabs.extra.KEEP_ALIVE", keepAliveIntent)
+    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+        channel.setMethodCallHandler(null)
+        channel = null as MethodChannel
```

Comment:
assigning null to channel causes a NPE

I was able to address the NPE with this change: https://github.com/greensopinion/flutter_web_auth/commit/f28e5a22b433b670ab5a9d598bad6e92489ae243